# Hypothesis Testing 1S2T - Call Center Process


In [1]:
import numpy as np
import pandas as pd
from scipy import stats
from scipy.stats import norm

### Sample Parameters: n=50, df=50-1=49, Mean1=4, SD1=3 
### 1-sample 2-tail ttest Assume Null Hypothesis Ho as Mean1 = 4 
### Thus, Alternate Hypothesis Ha as Mean1 ≠ 4

In [2]:
# For Day 10, Mean2 = 4.6 from the data table

# Find t_value = (Mean1 - Mean2)/(SD1/sqrt(n))
t_value1=(4-4.6)/(3/np.sqrt(50))
t_value1

-1.4142135623730943

In [3]:
# Find Probability P(X<t1)+P(X>t2) (Two-Tails Test values) so as to compare with Type-1 error α = 0.05

### For 2-tails; t1=-1.4 and t2=1.4 as t1=t2, t1+t2 = 2t1

In [4]:
p_value1=2*stats.t.cdf(-1.4142,df=49)
p_value1

0.1636259782698421

### As (p-value = 0.1636) > (α = 0.05); Accept Null Hypothesis Thus, Mean1 = 4 is correct by 16.36% probability

In [5]:
# For Day 15, Mean2 = 3.8 from the data table

# Find t_value = (Mean1 - Mean2)/(SD1/sqrt(n))
t_value2=(4-3.8)/(3/np.sqrt(50))
t_value2

0.4714045207910321

In [6]:
# Find Probability P(X<t1)+P(X>t2) (Two-Tails Test values) so as to compare with Type-1 error α = 0.05

### For 2-tails; t1=-0.47 and t2=0.47 as t1=t2, t1+t2 = 2t1

In [7]:
p_value2=2*stats.t.cdf(-0.47,df=49)
p_value2

0.640440149558531

In [8]:
# OR
p_value2=2*(1-stats.t.cdf(0.47,df=49))
p_value2

0.6404401495585312

### As (p-value = 0.64) > (α = 0.05); Accept Null Hypothesis Thus, Mean1 = 4 is correct by 64.04% probability

In [9]:
# For Day 13, Mean2 = 4.0 from the data table

# Find t_value = (Mean1 - Mean2)/(SD1/sqrt(n))
t_value3=(4-4)/(3/np.sqrt(50))
t_value3



0.0

In [10]:
# Find Probability P(X<t1)+P(X>t2) (Two-Tails Test values) so as to compare with Type-1 error α = 0.05

### For 2-tails; t1=-0 and t2=0 as t1=t2, t1+t2 = 2t1

In [12]:
p_value3=2*stats.t.cdf(-0,df=49)
p_value3

1.0

#### As (p-value = 1.0) > (α = 0.05); Accept Null Hypothesis Thus, Mean1 = 4 is correct by 100% probability If somehow, we happen to reject null hypthesis, the decision will be wrong by 100%